consensus/dummy/consensus.go

```diff
@@ -175,6 +196,12 @@ func (self *DummyEngine) verifyHeader(chain consensus.ChainHeaderReader, header
 	if err := self.verifyHeaderGasFields(config, header, parent, chain); err != nil {
 		return err
 	}
+	// Ensure that coinbase is valid if reward manager is enabled
+	if config.IsRewardManager(timestamp) {
+		if err := self.verifyCoinbase(config, header, parent, chain); err != nil {
+			return err
+		}
+	}
```

Comment:
just wanted to short-circuit it since we already check that in syntactic verify. but sure we can remove this condition.

Reply:
hmm apperantly we cannot always verify it here since it breaks unit tests :/. they don't expect any coinbase verification or set the coinbase to blackhole, so eventually they fail.
https://github.com/ava-labs/subnet-evm/actions/runs/3427844912/jobs/5711340511